src/components/TileGallery/StatefulTileGallery.jsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import get from 'lodash/get';
+import escapeRegExp from 'lodash/escapeRegExp';
+import { ButtonTypes } from 'carbon-components-react/lib/prop-types/types';
+
+import PageTitleBar from '../PageTitleBar';
+import Button from '../Button';
+
+import TileGallery from './TileGallery';
+import TileGallerySection from './TileGallerySection';
+import TileGalleryItem from './TileGalleryItem';
+import TileGallerySearch from './TileGallerySearch';
+import TileGalleryViewSwitcher from './TileGalleryViewSwitcher';
+
+const propTypes = {
+  /** Title of the page  */
+  title: PropTypes.string.isRequired,
+  /** Details about what the page shows */
+  description: PropTypes.node,
+  /** Has simple search capability */
+  hasSearch: PropTypes.bool,
+  /** Has toggle grid/list component */
+  hasSwitcher: PropTypes.bool,
+  /** Has button */
+  hasButton: PropTypes.bool,
+  /** i18n strings */
+  i18n: PropTypes.shape({
+    /** i18n strings for seach capability */
+    searchIconDescription: PropTypes.string,
+    searchPlaceHolderText: PropTypes.string,
+    searchCloseButtonText: PropTypes.string,
+    /** i18n strings for grid/list capability */
+    listText: PropTypes.string,
+    gridText: PropTypes.string,
+    descriptionMoreInfo: PropTypes.string,
+    arrowIconDescription: PropTypes.string,
+  }),
+  /** Custom className for component */
+  className: PropTypes.string,
+  galleryData: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      sectionTitle: PropTypes.string,
+      galleryItems: PropTypes.arrayOf(
+        PropTypes.shape({
+          title: PropTypes.string.isRequired,
+          description: PropTypes.string,
+          moreInfoLink: PropTypes.string,
+          icon: PropTypes.node,
```

Comment:
Another spot to mirror [Carbon's Icon proptype definition](icon: PropTypes.shape({
    width: PropTypes.string,
    height: PropTypes.string,
    viewBox: PropTypes.string.isRequired,
    svgData: PropTypes.object.isRequired,
  }),)